apps/server/src/modules/tldraw/utils/utils.ts

```diff
@@ -0,0 +1,252 @@
+import { setInterval, clearInterval } from 'timers';
+import * as Y from "yjs";
+import * as map from "lib0/dist/map.cjs";
+import * as awarenessProtocol from "y-protocols/dist/awareness.cjs";
+import * as encoding from "lib0/dist/encoding.cjs";
+import * as decoding from "lib0/dist/decoding.cjs";
+import * as syncProtocol from "y-protocols/dist/sync.cjs";
+import { isCallbackSet, callbackHandler } from "./callback";
+import * as debounce from "lodash.debounce";
+import { Configuration } from '@hpi-schul-cloud/commons';
+
+const CALLBACK_DEBOUNCE_WAIT = Configuration.get('FEATURE_TLDRAW_CALLBACK_DEBOUNCE_WAIT') as number ?? 2000;
+const CALLBACK_DEBOUNCE_MAX_WAIT = Configuration.get('FEATURE_TLDRAW_CALLBACK_DEBOUNCE_MAX_WAIT') as number ?? 10000;
+const pingTimeout = Configuration.get('FEATURE_TLDRAW_PING_TIMEOUT') as number ?? 30000;
+// disable gc when using snapshots!
+const gcEnabled = Configuration.get('FEATURE_TLDRAW_GC_ENABLED');
+
+enum wsStatesEnum {
+	wsReadyStateConnecting = 0,
+	wsReadyStateOpen = 1
+}
+
+enum wsMessageTypes {
+	messageSync = 0,
+	messageAwareness = 1
+}
+
+/**
+ * @type {{bindState: function(string,WSSharedDoc):void, writeState:function(string,WSSharedDoc):Promise<any>, provider: any}|null}
+ */
+let persistence: any = null;
+
+/**
+ * @param {{bindState: function(string,WSSharedDoc):void,
+ * writeState:function(string,WSSharedDoc):Promise<any>,provider:any}|null} persistence_
+ */
+export const setPersistence = persistence_ => {
+	persistence = persistence_;
+};
+
+/**
+ * @type {Map<string,WSSharedDoc>}
+ */
+const docs = new Map();
+
+/**
+ * @param {Uint8Array} update
+ * @param {any} origin
+ * @param {WSSharedDoc} doc
+ */
+const updateHandler = (update, origin, doc) => {
+	const encoder = encoding.createEncoder();
+	encoding.writeVarUint(encoder, wsMessageTypes.messageSync);
+	syncProtocol.writeUpdate(encoder, update);
+	const message = encoding.toUint8Array(encoder);
+	doc.conns.forEach((_, conn) => send(doc, conn, message));
+};
+
+export class WSSharedDoc extends Y.Doc {
+	private name: any;
```

Comment:
Does it need to be `any`? But also it looks like this field is not used anywhere in the first place.